LibGAP.jl/src/julia_to_gap.jl

```diff
@@ -72,29 +72,18 @@ function julia_to_gap(obj::Array{T,1}, recursive::Val{Recursive}=Val(false), rec
     ret_val = NewPlist(len)
     recursion_dict[obj] = ret_val
     for i in 1:len
-        current_obj = obj[i]
-        if haskey(recursion_dict,current_obj)
-            ret_val[i] = recursion_dict[current_obj]
-        else
-            if Recursive
-                current_converted = julia_to_gap(current_obj,recursive,recursion_dict)
-            else
-                current_converted = current_obj
-            end
-            recursion_dict[current_obj] = current_converted
-            ret_val[i] = current_converted
+        x = obj[i]
+        if Recursive
+            x = julia_to_gap(x, recursive, recursion_dict)
```

Comment:
Actually, I am now unsure whether this change is quite correct. It fixed a bug, but I think it's not quite what we intend. I'll elaborate in another comment on PR #173. Please do not merge until this is resolved.

Reply:
I have now rewritten this PR yet again, to match what I explained on PR #173